### ISSUE TITLE
pipeline_trigger.Read should handle 404, in case 3rd party removed trigger

### ISSUE THREAD

**Issue by romaninsh** [gitlabhq/terraform-provider-gitlab#590]
When terraform refreshes state and "pipeline_trigger" is missing (e.g. was removed manually), this causes terraform to abort with 404 error. Only solution is to manually update state.

This is caused by not handling "NotFound" error correctly:

 - pipeline_trigger - https://github.com/gitlabhq/terraform-provider-gitlab/blob/master/gitlab/resource_gitlab_pipeline_trigger.go#L68 "err" is always returned.
 - project_variable - has this error handled correctly: https://github.com/gitlabhq/terraform-provider-gitlab/blob/master/gitlab/resource_gitlab_project_variable.go#L133

### Terraform Version
master

### Affected Resource(s)
gitlab_pipeline_trigger

### Terraform Configuration Files
```hcl
resource "gitlab_pipeline_trigger" "trigger" {
  project     = "project-x"
  description = "Trigger"
}
```

### Debug Output
Please provider a link to a GitHub Gist containing the complete debug output: https://www.terraform.io/docs/internals/debugging.html. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

### Panic Output
Error: GET https://git.oxfordnanolabs.local/api/v4/projects/project-x/triggers/5: 404 {message: 404 Not found}

### Expected Behavior
```
if errors.Is(err, errPipelineTriggerNotExist) {
			log.Printf("[DEBUG] read gitlab pipeline trigger %q was not found", d.Id())
			d.SetId("")
			return nil
		}
```

### Actual Behavior
`terraform plan` fails, without even suggesting what state ID caused it, so if you have multiple pipeline trigger - good luck trying to find which one has gone missing.

### Steps to Reproduce
> terraform apply
> remove "Trigger" manually under Settings, CI/CD, Triggers
> terraform plan

Plan Fails.

### References
 - pipeline_trigger - https://github.com/gitlabhq/terraform-provider-gitlab/blob/master/gitlab/resource_gitlab_pipeline_trigger.go#L68 
 - project_variable - has this error handled correctly: https://github.com/gitlabhq/terraform-provider-gitlab/blob/master/gitlab/resource_gitlab_project_variable.go#L133

**Comment by romaninsh** [comment]
@armsnyder - I wonder if you could help by looking into this? I see you have done some great work around project_variable.

**Comment by armsnyder** [comment]
Hi @romaninsh. 😄  It looks like the proper logic is being added as part of this PR: #318

**Comment by timofurrer** [comment]
It's been fixed in #318 🎉
